content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,425 @@
+## Enabling DVC pipelines for checkpoint experiments
+
+In the `dvc.yaml` file, we need to add a few things to get our checkpoints and
+metrics tracking setup. First we'll add metrics tracking to our experiments. At
+the end of the file, add the following lines.
+
+```yaml
+live:
+  dvclive:
+    summary: true
+    html: true
+```
+
+Your `dvc.yaml` should look similar to this now.
+
+```yaml
+---
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+live:
+  dvclive:
+    summary: true
+    html: true
+```
+
+Now we'll set up checkpoints. Below the `model.pt`, add the following code:
+`checkpoint: true`. This enables checkpoints at the pipeline level. Your
+`dvc.yaml` should look like this now.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+          template: confusion
+          x: actual
+          y: predicted
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`--checkpoint` option of the `dvc stage add` command defines the checkpoint file
+or directory. This will create a new stage with checkpoints enabled, but does
+not update existing stages.
+
+If you want to update an existing stage, adding the `-f` option to the
+`dvc stage add --checkpoint` command will force an overwrite of the current
+stage and update it with checkpoints enabled.
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+To run this code, run the `dvc exp run` command and it will start training the
+model. In the terminal, you'll see the training code being executed and you'll
+see the metrics being recorded.
+
+```dvc
+Running stage 'train':
+> python train.py
+Epoch 1: loss=0.4989388585090637
+Epoch 1: acc=0.8481
+Epoch 2: loss=0.3318292796611786
+Epoch 2: acc=0.9037
+Epoch 3: loss=0.21992072463035583
+```
+
+Now we need to enable checkpoints at the pipeline level. We are interested in
+tracking the metrics along with each checkpoint, so we'll need to add a few
+lines of code.
+
+In the `train.py` file, update the following lines of code in the `main` method
```

Comment:
`import dvclive` is imported but not used :) It is better to remove it and ask a user to import.

Reply:
Also, for some reason train.py does not work from outside of dvc (works just fine from DVC):
```bash
$ python train.py
Traceback (most recent call last):
  File "train.py", line 121, in <module>
    main()
  File "train.py", line 97, in main
    mnist_train = torchvision.datasets.MNIST("data")
  File "/Users/dmitry/mlops-sf/dvc-checkpoints-mnist/.venv/lib/python3.7/site-packages/torchvision/datasets/mnist.py", line 88, in __init__
    raise RuntimeError('Dataset not found.' +
RuntimeError: Dataset not found. You can use download=True to download it
```